### DISCUSSION TITLE
App Dir - Full Page Refresh results in Blank Page

### DISCUSSION THREAD

**Discussion by toddgeist** [vercel/community#1729] [Help] [answered]
### Summary

I have been struggling with a next 13 app directory issue for a few days. Everything works fine on locally but, after deploying to vercel I get react server component errors, and a blank page when doing a full page refresh or navigating directly to a post.  I have tried the latest stable 13.2.3, and the latest canary 13.2.4-canary.6. The error message has changed a bit with the different canaries.  Currently I get this error below with the latest 13.2.4-canary.6. I can see this in the Vercel logs.

Note the error is happening in `react-server-dom-webpack-client.edge.production.min.js:`. Despite the fact that I haven't enabled edge functions anywhere, that I am aware of.

I have not been able to reproduce this in a small repo yet.

Any ideas?

### Additional information

```
SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at T (/var/task/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:21:53)
    at c (/var/task/node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.production.min.js:23:307)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by amyegan** [comment]
Hi @toddgeist. Do you have a public repo you can share here?

**Comment by toddgeist** [comment]


**Reply by toddgeist** [reply]
Sorry I needed to take this private again.  I thought this was the backchannel support :-(

**Reply by amyegan** [reply]
No problem if you need to keep the repo away from public view. I see you opened a support case as well, so you'll be able to communicate with the team more privately there. I deleted your previous message to remove the repo link from this discussion.

Are there any more clues in your [Runtime Logs](https://vercel.com/docs/concepts/observability/runtime-logs#) or in your browser's dev tools? I recall getting an error like this before when an API call returned an error page instead of JSON data. It may be worth double checking the project's [environment variables](https://vercel.com/docs/concepts/projects/environment-variables). Or try temporarily logging or try/catching any related API calls. Invalid tokens and CORS problems seem to be the common causes of mysterious errors.

**Reply by toddgeist** [reply]
the runtime logs give pretty much the same error.  Just with a little more detail about what file it is happening.

Client side nav works fine, server side is completely busted

**Reply by toddgeist** [reply]
Probably the most frustrating thing I have encountered with the app directory so far.

**Reply by toddgeist** [reply]
the only other odd thing I can say about it is getting rid of one the layout inside the catch all folder fixes the problem.  I need that layout though.

**Comment by toddgeist** [comment]
all I have to do is add a simple Layout and it breaks
![tg-2023-03-08 at 16 49 16](https://user-images.githubusercontent.com/316792/223886507-1f3e4251-71ef-47f4-b090-972e629d57ad.png)

If I get rid of that. Server Side Navigation works.

**Reply by toddgeist** [reply, marked as answer]
it appears that the problem goes away if the catch all route is not at the root.
![tg-2023-03-08 at 17 35 01](https://user-images.githubusercontent.com/316792/223892787-178fd072-0016-49b2-8d34-573ce13b4405.png)

**Comment by swarnava** [comment]
Hi, I can see that your latest build has successfully deployed. Let us know if you are still facing issue.
